codec-http2/src/main/java/io/netty/handler/codec/http2/AbstractHttp2StreamFrame.java

```diff
@@ -50,11 +45,15 @@ public boolean equals(Object o) {
             return false;
         }
         Http2StreamFrame other = (Http2StreamFrame) o;
-        return streamId == other.streamId();
+        return stream == other.stream() || (stream != null && stream.equals(other.stream()));
     }
 
     @Override
     public int hashCode() {
```

Comment:
I vote for keeping them now, and re-evaluate in a followup PR. I'm guessing removing these will have an impact on unit tests which might get messy.